src/sqlfluff/core/rules/std.py

```diff
@@ -3244,3 +3244,60 @@ def _eval(self, segment, **kwargs):
                 self.violation_buff[-1].fixes = fixes
 
         return self.violation_buff or None
+
+
+@std_rule_set.document_fix_compatible
+@std_rule_set.register
+class Rule_L035(BaseCrawler):
+    """Do not specify "else null" in a case when statement (redundant).
+
+    | **Anti-pattern**
+
+    .. code-block:: sql
+
+        select
+            case
+                when name like '%cat%' then 'meow'
+                when name like '%dog%' then 'woof'
+                else null
+            end
+        from x
+
+
+    | **Best practice**
+    |  Omit "else null"
+
+    .. code-block:: sql
+
+        select
+            case
+                when name like '%cat%' then 'meow'
+                when name like '%dog%' then 'woof'
+            end
+        from x
+
+    """
+
+    def _eval(self, segment, **kwargs):
+        if segment.is_type("case_expression"):
```

Comment:
Neat way of working the logic here, can you write a brief description of how it works for future users in the docstring of `_eval`?

In particular noting that there are two phases to operation, first we look for `ELSE` (indicated by `fixes` being []), then we keep generating deletions up to finding `NULL` in the immediate child segments. If `NULL` is never found, the fixes (and therefore the violation) are never reported, but if `NULL` is found then the violation and all the associated deletions are also reported.

Reply:
I went into detail, step by step in the docstring of the _eval, can you have a look ?